extensions/spark/kyuubi-spark-connector-hive/src/main/scala/org/apache/kyuubi/spark/connector/hive/HiveTableCatalog.scala

```diff
@@ -0,0 +1,410 @@
+      case _ =>
+    }
+    conf
+  }
+
+  private lazy val sparkConf: SparkConf = {
+    val conf = sparkSession.sparkContext.getConf
+    catalogOptions.asScala.foreach {
+      case (k, v) => conf.set(k, v)
+    }
+    conf
+  }
+
+  def hadoopConfiguration(): Configuration = hadoopConf
+
+  override def name(): String = {
+    require(catalogName != null, "The Hive table catalog is not initialed")
+    catalogName
+  }
+
+  override def initialize(name: String, options: CaseInsensitiveStringMap): Unit = {
+    assert(catalogName == null, "The Hive table catalog is already initialed.")
+    assert(
+      conf.getConf(CATALOG_IMPLEMENTATION) == "hive",
+      s"Require setting ${CATALOG_IMPLEMENTATION.key} to `hive` to enable hive support.")
+    catalogName = name
+    catalogOptions = options
+    catalog = new HiveSessionCatalog(
+      externalCatalogBuilder = () => externalCatalog,
+      globalTempViewManagerBuilder = () => sparkSession.sharedState.globalTempViewManager,
+      metastoreCatalog = new HiveMetastoreCatalog(sparkSession),
+      functionRegistry = sessionState.functionRegistry,
+      tableFunctionRegistry = sessionState.tableFunctionRegistry,
+      hadoopConf = hadoopConf,
+      parser = sessionState.sqlParser,
+      functionResourceLoader = sessionState.resourceLoader)
+  }
+
+  /**
+   * A catalog that interacts with external systems.
+   */
+  lazy val externalCatalog: ExternalCatalogWithListener = {
+    val externalCatalog = new HiveExternalCatalog(sparkConf, hadoopConf)
+
+    // Wrap to provide catalog events
+    val wrapped = new ExternalCatalogWithListener(externalCatalog)
+
+    // Make sure we propagate external catalog events to the spark listener bus
+    wrapped.addListener((event: ExternalCatalogEvent) => postExternalCatalogEvent(sc, event))
+
+    wrapped
+  }
+
+  override val defaultNamespace: Array[String] = Array("default")
+
+  override def listTables(namespace: Array[String]): Array[Identifier] = {
+    namespace match {
+      case Array(db) =>
+        catalog
+          .listTables(db)
+          .map(ident => Identifier.of(ident.database.map(Array(_)).getOrElse(Array()), ident.table))
+          .toArray
+      case _ =>
+        throw new NoSuchNamespaceException(namespace)
+    }
+  }
+
+  override def loadTable(ident: Identifier): Table = {
+    HiveTable(sparkSession, catalog.getTableMetadata(ident.asTableIdentifier), this)
+  }
+
+  override def createTable(
+      ident: Identifier,
+      schema: StructType,
+      partitions: Array[Transform],
+      properties: util.Map[String, String]): Table = {
+    val (partitionColumns, maybeBucketSpec) = convertTransforms(partitions.toSeq)
+    val provider = properties.getOrDefault(TableCatalog.PROP_PROVIDER, conf.defaultDataSourceName)
+    val tableProperties = properties.asScala
+    val location = Option(properties.get(TableCatalog.PROP_LOCATION))
+    val storage = DataSource.buildStorageFormatFromOptions(toOptions(tableProperties.toMap))
```

Comment:
I think create table a.b.c (...) stored as orc should be also supported in hive dsv2, shall we?